app/task_routes.py

```diff
@@ -0,0 +1,140 @@
+# import requests
+import os
+from dotenv import load_dotenv
+from app.slack_api import notify_slack
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+load_dotenv()
+
+def validate_task(task_id):
+    try:
+        task_id = int(task_id)
+    except:
+        abort(make_response({"message": f"task {task_id} is invalid"}, 400))
+
+    task = Task.query.get(task_id)
+    
+    if not task:
+        abort(make_response({"details": f"Task {task_id} not found"}, 404))
+    
+    return task
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    
+    if "description" not in request_body or "title" not in request_body:
+        abort(make_response({"details": "Invalid data"}, 400))
+    
+    new_task = Task(title=request_body["title"],
+                    description=request_body["description"])
+    
+    
+    db.session.add(new_task)
+    db.session.commit()
+    
+    return {
+        "task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": new_task.is_complete
+        }
+    }, 201
+    
+    # return make_response(jsonify(f"Task {new_task.title} successfully created"), 201)
+
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+        
+    sort_query = request.args.get("sort")
+    if sort_query:
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title)
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+    
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+        
+    return jsonify(tasks_response)
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_task(task_id)
+    
+    return {"task": task.to_dict()}
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_task(task_id)
+    
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body:
+        abort(make_response({"details": f"Invalid data"}, 400))
+    
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    
+    db.session.commit()
+    
+    return {"task": task.to_dict()}
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_task(task_id)
+    
+    db.session.delete(task)
+    db.session.commit()
+    
+    return make_response({"details": f'Task {task.task_id} "{task.title}" successfully deleted'})
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = validate_task(task_id)
+    
+    task.completed_at = datetime.utcnow()
+    task.is_complete = True
+    
+    db.session.commit()
+
+    slack_token = os.environ.get("THE_SLACK_TOKEN")
```

Comment:
Not a big issue but it's a convention to avoid using articles in environment variable names. In your `slack_api.py` file, there's a commented out call to get an environment variable `SLACK_BOT_USER_OAUTH_TOKEN`, which would be a great choice for the name!